src/testing/mocks/middleware/intersection.ts

```diff
@@ -39,12 +39,12 @@ export function createIntersectionMock() {
 		}
 	};
 
-	function mockIntersection(): MiddlewareResult<any, any, any>;
+	function mockIntersection(): MiddlewareResult<any, any, any, any>;
```

Comment:
can we add a convenience type for this? it's used quite a lot in the mocks, and presumably would be also for a user writinga mock?

Reply:
I added a `DefaultMiddlewareResult` interface that represents `MiddlewareResult<any, any, any, any>`